### PR TITLE
Keep GitHub Actions up to date with GitHub's Dependabot

### PR DESCRIPTION
**Pull request by cclauss**
Automation that fixes software supply chain warnings like at the bottom right of
https://github.com/exercism/free-pascal/actions/runs/8474048489

* https://docs.github.com/en/code-security/dependabot/working-with-dependabot/keeping-your-actions-up-to-date-with-dependabot
* https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file#package-ecosystem

### CODE REVIEW

**Comment by cclauss** [comment]
@ErikSchierboom
* #18
* #19

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+# https://docs.github.com/en/code-security/dependabot/working-with-dependabot/keeping-your-actions-up-to-date-with-dependabot
+# https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file#package-ecosystem
+version: 2
+updates:
+  - package-ecosystem: github-actions
+    directory: /
+    groups:
+      github-actions:
+        patterns:
+          - "*"  # Group all Actions updates into a single larger pull request
```
Comment (ErikSchierboom) [inline .github/dependabot.yml:11]: We usually don't do this as one update could cause things to fail and then the other updates can't be merged.

BTW, this is our sort of default dependabot config: https://github.com/exercism/prolog/blob/main/.github/dependabot.yml

**Comment by cclauss** [comment]
```
Report
Creating test report Junit Test Results
Error: HttpError: Resource not accessible by integration
```

File: .github/dependabot.yml (hunk 1 above)
Comment (cclauss) [reply .github/dependabot.yml:11]: Why is the default config not running on this repo?

**Comment by cclauss** [comment]
Let's try with the current version plus
* https://github.com/dorny/test-reporter?tab=readme-ov-file#recommended-setup-for-public-repositories

**Review by ErikSchierboom — APPROVED** [review]
LGTM

**Comment by cclauss** [comment]
Still failing...  Let's get the test passing before we merge.

**Comment by cclauss** [comment]
OK.  Once `.github/workflows/test-report.yml` is merged to the default branch this should work.

**Pull request merged**